Adding support for SAS Token Generation

ksaye opened the pull request:
Azure IoT Hub uses a Sha256 has as opposed to a password.  This change enables the admin to add the 'Preshared Key' and lets Tasmota calculate the SAS token, vs the original changes where the end user had to calculate the SAS Token.

## Description:

**Related issue (if applicable):** fixes #<Tasmota issue number goes here>

## Checklist:
  - [X] The pull request is done against the latest development branch
  - [X] Only relevant files were touched
  - [X] Only one feature/fix was added per PR and the code change compiles without warnings
  - [X] The code change is tested and works on Tasmota core ESP8266 V.2.7.4.9
  - [ ] The code change is tested and works with core ESP32 V.1.0.6
  - [X] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

ksaye committed:
Adding support for SAS Token Generation

Azure IoT Hub uses a Sha256 has as opposed to a password.  This change enables the admin to add the 'Preshared Key' and lets Tasmota calculate the SAS token, vs the original changes where the end user had to calculate the SAS Token.

ksaye committed:
Moving the SSL and Base64 to global

ascillato commented:
Hi,

Why you add that last change in 5b90754 ? Seems that was ok as it was before that wasn't including libraries that were not used.

ascillato commented on tasmota/xdrv_02_mqtt.ino:
```diff
@@ -23,9 +23,9 @@
-#ifdef USE_MQTT_AZURE_IOT
 #include <bearssl\bearssl.h>
 #include <base64.hpp>
+#ifdef USE_MQTT_AZURE_IOT
```
Why is this last change used for?

ksaye replied:
I received the failed to build message when they were not global, as the method azurePreSharedKeytoSASToken was global and not under the #ifdef

ascillato reviewed (changes requested):
Why is this last change used for?

Seems that was ok as it was before that wasn't including libraries that were not used.

ksaye committed:
Moving includes back to local

Moving the includes back under the '#ifdef USE_MQTT_AZURE_IOT', so the build process will complete for all builds.

ascillato commented on tasmota/xdrv_02_mqtt.ino:
```diff
@@ -24,6 +24,8 @@
 #endif
 
 #ifdef USE_MQTT_AZURE_IOT
+#include <bearssl\bearssl.h>
```
What do you think of using instead the library that Tasmota already has?

```cpp
#include <t_bearssl.h>
```

ascillato commented:
I have tested your PR and works fine. Great Job!

What do you think of using the bearssl library that already Tasmota has?

So, instead of using:
```cpp
#include <bearssl\bearssl.h>
```
Change to use:
```cpp
#include <t_bearssl.h>
```

ascillato2 committed:
Use t_bearssl.h instead of bearssl.h

ascillato2 commented:
I end up changing that to t_bearssl.h
